src/endpoints/getEvents.ts

```diff
@@ -0,0 +1,85 @@
+import { EventResult } from 'models/EventResult';
+import { SimpleEvent } from 'models/SimpleEvent';
+import { EventSize } from 'enums/EventSize';
+
+export const getEvents = (config: HLTVConfig) => async ({
+    size
+}: {
+    size?: EventSize
+}): Promise<EventResult[]> => {
+    const $ = await fetchPage(`${config.hltvUrl}/events`, config.loadPage);
+
+    let events = [] as EventResult[];
+
+    toArray($('.events-month')).map(event => {
+        let monthEvents = [] as SimpleEvent[];
+        let monthName   = event.find('.standard-headline').text();
+
+        switch(size) {
+            case EventSize.Small:
+                monthEvents = parseEvents(toArray(event.find('a.small-event')), EventSize.Small);
+            break;
+
+            case EventSize.Big:
+                monthEvents = parseEvents(toArray(event.find('a.big-event')), EventSize.Big);
+            break;
+
+            default:
+                monthEvents = parseEvents(toArray(event.find('a.big-event'))).concat( parseEvents(toArray(event.find('a.small-event'))) );
+            break;
+        }
+
+        events.push({
+            month: monthName,
+            events: monthEvents
+        });
+    });
+
+    return events;
+}
+
+const parseEvents = (eventsToParse, size?:EventSize) => {
+    let events = [] as SimpleEvent[];
+
+    let dateSelector, nameSelector, locationSelector = '';
+
+    if(size == EventSize.Small) {
+        dateSelector        = '.eventDetails .col-desc span[data-unix]';
+        nameSelector        = '.col-value .text-ellipsis';
+        locationSelector    = '.smallCountry img';
+    } else {
+        dateSelector        = 'span[data-unix]';
+        nameSelector        = '.big-event-name';
+        locationSelector    = '.location-top-teams img';
+    }
+
+    eventsToParse.forEach((eventEl) => {
+        let dateStart   = eventEl.find(dateSelector).eq(0).data('unix');
+        let dateEnd     = eventEl.find(dateSelector).eq(1).data('unix');
+        let teams       = '0';
+        let prizePool   = '';
+
+        if(size == EventSize.Small) {
+            teams       = eventEl.find('.col-value').eq(1).text();
+            prizePool   = eventEl.find('.prizePoolEllipsis').text();
+        } else {
+            teams       = eventEl.find('.additional-info tr').eq(0).find('td').eq(2).text();
+            prizePool   = eventEl.find('.additional-info tr').eq(0).find('td').eq(1).text();
+        }
+
+        events.push({
+            id: Number(eventEl.attr('href').split('/')[2]),
+            name: eventEl.find(nameSelector).text(),
+            dateStart: dateStart ? Number(dateStart) : undefined,
+            dateEnd: dateEnd ? Number(dateEnd) : undefined,
+            prizePool: prizePool,
+            teams: teams.length ? Number(teams) : undefined,
+            location: eventEl.find(locationSelector).prop('title'),
+            host: size == EventSize.Small ? eventEl.find('table tr').eq(0).find('td').eq(3).text() : undefined
```

Comment:
There's a bug in here - if I don't pass in a size I still want the host to be parsed.

Reply:
Should be good now